test/Main.purs

```diff
@@ -26,29 +29,24 @@ derive newtype instance ringTestRational :: Ring TestRational
 derive newtype instance semiringTestRational :: Semiring TestRational
 derive newtype instance divisionRingTestRational :: DivisionRing TestRational
 
-int :: Gen Int
-int = chooseInt (-999) 999
-
-nonZeroInt :: Gen Int
-nonZeroInt = int `suchThat` notEq 0
-
-newtype SmallInt = SmallInt Int
+bigint :: Gen BigInt
+bigint = fromInt <$> chooseInt (-999) 999
```

Comment:
You can now remove the `chooseInt` and just use `arbitrary` to get an arbitrary `Int`. I think the `SmallInt` was there because of the overflow.

Reply:
good point!